src/instructlab/utils.py

```diff
@@ -556,15 +566,89 @@ def split_hostport(hostport: str) -> tuple[str, int]:
     return hostname, port
 
 
-def convert_messages_to_legacy_dataset(
+def is_pretraining_dataset(ds: List[MessageSample]) -> bool:
+    """
+    Determines whether or not the given dataset is the phase007 pretraining
+    dataset.
+    """
+    if not ds:
+        return False
+    sample = ds[0]
+    return any(m["role"] == "pretraining" for m in sample["messages"])
+
+
+def get_user_assistant_from_pretraining(pretraining: str) -> Tuple[str, str]:
+    """
+    Given a pretraining message sample which contains the tokens '<|user|>' and '<|assistant|>',
+    return the strings corresponding to the user & assistant inputs.
+
+    This function assumes that a given message begins with the '<|user|>' token
```

Comment:
Is this enforced somewhere in the format definition? Or is just implementation detail that may change later? (Could you please share a link to where the format is defined? Maybe add a link to the comment?)

Reply:
It's not necessarily enforced, but there's no logical reason for anything to exist before `<|user|>` since we're providing that as a separate message.

We can change this to instead extract all of the text between `<|user|>` and `<|assistant|>` rather than assuming that `<|user|>` is the very first token of the string.